services/elastic_ip_manager/src/main/java/com/futurewei/alcor/elasticipmanager/service/implement/ElasticIpServiceImpl.java

```diff
@@ -258,6 +269,56 @@ public ElasticIpInfo updateElasticIp(ElasticIpInfo request) throws Exception {
         return new ElasticIpInfo(eip);
     }
 
+    private Map<String, ElasticIp> getElasticIpsByPortId(String projectId, String portId) {
+
+        Map<String, Object[]> filter = new HashMap<>();
+        filter.put("project_id", new Object[] {projectId});
+        filter.put("port_id", new Object[] {portId});
+
+        return elasticIpRepo.findAllItems(filter);
+    }
+
+    private String getAssociatedPortIp(String projectId, String portId, Integer ipVersion, String ipAddress)
+            throws Exception {
+        PortManagerProxy portManagerProxy = new PortManagerProxy(projectId);
+        PortEntity port = portManagerProxy.getPortById(portId);
+
+        List<PortEntity.FixedIp> fixedIps = port.getFixedIps();
+        List<String> fixedIpv4List = new ArrayList<>();
+        for (PortEntity.FixedIp fixedIp: fixedIps) {
+            if (Ipv4AddrUtil.formatCheck(fixedIp.getIpAddress())) {
+                fixedIpv4List.add(fixedIp.getIpAddress());
+            }
+        }
+
+        if (fixedIpv4List.isEmpty()) {
+            throw new ElasticIpPipNotFound();
+        }
+
+        String associatedIp;
+        if (ipAddress != null) {
+            if (fixedIpv4List.contains(ipAddress)) {
+                associatedIp = ipAddress;
+            } else {
+                throw new ElasticIpPipNotFound();
+            }
+        } else {
+            if (fixedIpv4List.size() > 1) {
+                throw new ElasticIpMultipleFixedIpFound();
+            }
+            associatedIp = fixedIpv4List.get(0);
+        }
+
+        Map<String, ElasticIp> associatedEips = this.getElasticIpsByPortId(projectId, projectId);
+        for (ElasticIp eip: associatedEips.values()) {
+            if (associatedIp.equals(eip.getPrivateIp())) {
+                throw new ElasticIpAssociateConflict();
```

Comment:
It would be a good idea to include the conflicted ip in the exception message.

Reply:
OK, I will try

Reply:
Thanks.

Reply:
Done